src/main/resources/db/migration/V1__tables.sql

```diff
@@ -0,0 +1,58 @@
+BEGIN;
+
+DROP TABLE IF EXISTS objects CASCADE;
+DROP TABLE IF EXISTS object_log CASCADE;
+DROP TABLE IF EXISTS versions CASCADE;
+
+CREATE TABLE objects
+(
+    id         BIGSERIAL PRIMARY KEY,
+    hash       CHAR(64) NOT NULL,
+    url        TEXT     NOT NULL,
+    client_id  TEXT     NOT NULL,
+    content    BYTEA    NOT NULL,
+    is_deleted BOOLEAN  NOT NULL DEFAULT FALSE
+);
+
+CREATE TABLE object_log
+(
+    id            BIGSERIAL PRIMARY KEY,
+    operation     CHAR(3) NOT NULL,
+    new_object_id BIGINT,
+    old_object_id BIGINT,
+    CHECK (operation IN ('INS', 'UPD', 'DEL')),
+    CHECK (
+            operation = 'INS' AND old_object_id IS NULL AND new_object_id IS NOT NULL OR
+            operation = 'UPD' AND new_object_id IS NOT NULL AND new_object_id IS NOT NULL OR
+            operation = 'DEL' AND old_object_id IS NOT NULL AND new_object_id IS NULL
+        )
+);
+
+CREATE TABLE versions
+(
+    id                BIGSERIAL PRIMARY KEY,
+    session_id        TEXT   NOT NULL,
+    serial            BIGINT NOT NULL,
+    last_log_entry_id BIGINT NOT NULL,
+    snapshot_hash     CHAR(64),
```

Comment:
Much more compact in bytes instead of as hex string

Reply:
I am thinking about removing the hash field and adding a functional index on it, like 
`create index bla on objects(sha256(hash))`
it would solve both compactness thing and automate the management of it.

Reply:
Nah, will add it as a separate refactoring later, too much of a hassle.